services/QuillLMS/app/models/activity_session.rb

```diff
@@ -389,13 +383,8 @@ def self.save_concept_results(classroom_unit_id, activity_id, concept_results)
     end
   end
 
-  def self.delete_activity_sessions_with_no_concept_results(classroom_unit_id, activity_id)
+  def self.delete_activity_sessions_with_no_concept_results(activity_sessions)
     incomplete_activity_session_ids = []
-    activity = Activity.find_by_id_or_uid(activity_id)
-    activity_sessions = ActivitySession.where(
-      classroom_unit_id: classroom_unit_id,
-      activity: activity
-    )
     activity_sessions.each do |as|
```

Comment:
Optional nit: you could make the previous code a little tighter with `select` + `map`